src/content/lodash/object/merge/vanilla.js

```diff
@@ -0,0 +1,49 @@
+function merge (...args) {
+  const result = {}
+  if (!args || !args.length) {
+    return result
+  }
+
+  args.forEach(currentArg => {
+    if (currentArg) {
+      Object.keys(currentArg).forEach(key => {
+        if (isObject(currentArg[key]) && result[key] && isObject(result[key])) {
```

Comment:
I don't think you need `&& result[key]` because `isObject` is already checking the  `data` argument

Reply:
You are right, I have removed this check

Reply:
cool, I won't have more time to look into it today (sorry about that), but I was wondering if you'd be keen on trying to write some tests too. I can look into that myself if you are not comfortable with it.

Reply:
Sure, will write test cases

Reply:
amazing, let me know if you need some guidance: I've been working on this alone so far and I'm not sure how clear it might be for other people :)